src/connection.rs

```diff
@@ -103,18 +103,28 @@ impl Connection {
                                 .start_send(tokio_tungstenite::tungstenite::Message::text(json_msg))
                             {
                                 Ok(_) => {
-                                    assert!(matches!(
-                                        self.pending_requests.insert(msg.reference, sender),
-                                        None
-                                    ));
+                                    if self
+                                        .pending_requests
+                                        .insert(msg.reference, sender)
+                                        .is_some()
+                                    {
+                                        log::debug!("Replaced request with same ref");
```

Comment:
Maybe warn? Just thinking, can this happen in a correct-usage of the lib? Also, maybe add the ref in the log?

Reply:
Would only happen if there was a bug in the lib

Reply:
Aaaaa, gg. Maybe error then?